pkg/engine2/path_selection/candidate_validity.go

```diff
@@ -247,31 +249,40 @@ func (d downstreamChecker) makeValid(resource, operationResource *construct.Reso
 	if err != nil {
 		return err
 	}
+	// include the operation resource in downstreams in case it also can be assigned to the target property
+	downstreams = append(downstreams, operationResource.ID)
 	cfgCtx := solution_context.DynamicCtx(d.ctx)
 
-	assign := func(resource construct.ResourceId, property string) (bool, error) {
+	assign := func(r construct.ResourceId, property string) (bool, error) {
 		var errs error
-		rt, err := d.ctx.KnowledgeBase().GetResourceTemplate(resource)
+		rt, err := d.ctx.KnowledgeBase().GetResourceTemplate(r)
 		if err != nil || rt == nil {
 			return false, fmt.Errorf("error getting resource template for resource %s: %w", resource, err)
 		}
 		p := rt.Properties[property]
 		for _, downstream := range downstreams {
-			val, err := knowledgebase.TransformToPropertyValue(resource, property, downstream, cfgCtx,
-				knowledgebase.DynamicValueData{Resource: resource})
+			val, err := knowledgebase.TransformToPropertyValue(r, property, downstream, cfgCtx,
+				knowledgebase.DynamicValueData{Resource: r})
 			if err != nil || val == nil {
 				continue // Becuase this error may just mean that its not the right type of resource
 			}
-			res, err := d.ctx.RawView().Vertex(resource)
-			if err != nil {
-				errs = errors.Join(errs, fmt.Errorf("error getting resource %s: %w", resource, err))
-				continue
+			// We need to check if the current resource is what we are operating on and if so not search our raw view
+			// this is because it could be a phantom resource
+			var currRes *construct.Resource
+			if resource.ID.Matches(r) {
+				currRes = resource
+			} else {
+				currRes, err = d.ctx.RawView().Vertex(r)
+				if err != nil {
+					errs = errors.Join(errs, fmt.Errorf("error getting resource %s: %w", resource, err))
+					continue
+				}
 			}
 			if p.IsPropertyTypeScalar() {
-				res.SetProperty(property, downstream)
+				currRes.SetProperty(property, downstream)
 				return true, errs
 			} else {
-				res.AppendProperty(property, downstream)
+				currRes.AppendProperty(property, downstream)
 				return true, errs
```

Comment:
Feels like this could be simplified by adding `Append` to scalar types that just do `Set`.

Reply:
yeah could add that as well if we want